src/Microsoft.TestPlatform.CoreUtilities/Output/OutputExtensions.cs

```diff
@@ -21,38 +21,44 @@ public static class OutputUtilities
         /// </summary>
         /// <param name="output">Output instance the method is being invoked with.</param>
         /// <param name="format">Format string for the error message.</param>
+        /// <param name="foregroundColor">Color in which text prints.</param>
         /// <param name="args">Arguments to format into the format string.</param>
-        public static void Error(this IOutput output, string format, params object[] args)
+        public static void Error(this IOutput output, string format, ConsoleColor foregroundColor=ConsoleColor.Red, params object[] args)
         {
-            using (new ConsoleColorHelper(ConsoleColor.Red))
+            ConsoleColorHelper.SetColorForAction(foregroundColor, () =>
             {
                 Output(output, OutputLevel.Error, DefaultFormat, format, args);
-            }
+            });
         }
 
         /// <summary>
         /// Output a warning message.
         /// </summary>
         /// <param name="output">Output instance the method is being invoked with.</param>
         /// <param name="format">Format string for the warning message.</param>
+        /// <param name="foregroundColor">Color in which text prints.</param>
         /// <param name="args">Arguments to format into the format string.</param>
-        public static void Warning(this IOutput output, string format, params object[] args)
+        public static void Warning(this IOutput output, string format, ConsoleColor foregroundColor=ConsoleColor.Yellow, params object[] args)
         {
-            using (new ConsoleColorHelper(ConsoleColor.Yellow))
+            ConsoleColorHelper.SetColorForAction(foregroundColor, () =>
             {
                 Output(output, OutputLevel.Warning, DefaultFormat, format, args);
-            }
+            });
         }
 
         /// <summary>
         /// Output a informational message.
         /// </summary>
         /// <param name="output">Output instance the method is being invoked with.</param>
         /// <param name="format">Format string for the informational message.</param>
+        /// <param name="foregroundColor">Color in which text prints.</param>
         /// <param name="args">Arguments to format into the format string.</param>
-        public static void Information(this IOutput output, string format, params object[] args)
+        public static void Information(this IOutput output, string format, ConsoleColor foregroundColor=ConsoleColor.White, params object[] args)
```

Comment:
We don't need to give the color for Error and Warning.
Probably, we can have overload for Information that takes in color ?